### PR TITLE
added Seurat RPCA mode

### PR DESCRIPTION
**Pull request by LuckyMD**
Add Seurat alignment mode using RPCA to find anchors.

### CODE REVIEW

**Review by danielStrobl — CHANGES_REQUESTED** [review]
This doesn't seem to work, I get this error:
```
Loading required package: Seurat
Error in PrepDR(object = object, features = features, verbose = verbose) : 
  Data has not been scaled. Please run ScaleData and retry
Calls: runSeuratRPCA ... RunPCA -> RunPCA.Seurat -> RunPCA -> RunPCA.Assay -> PrepDR
Execution halted
```

**Comment by danielStrobl** [comment]
Unfortunately this still doesn't work, I now get this error:
```
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 't': subscript out of bounds
```

File: R/integration.R [hunk 1]
```diff
@@ -39,6 +40,46 @@ runSeurat = function(data, batch, hvg=2000) {
         	   verbose = T)
 	  return(integrated)
 }
+
+runSeuratRPCA = function(data, batch, hvg=2000) {
+	  require(Seurat)
+	  batch_list = SplitObject(data, split.by = batch)
+
+          features <- SelectIntegrationFeatures(batch_list)
```
Comment (LuckyMD) [inline R/integration.R:48]: this we want to avoid. So can we just put `features <- hvg` or something here for the time being?

**Review by LuckyMD — COMMENTED** [review]
we need to use our HVG features... maybe draft sth that might do that and we can get input from Jain on how this can be done.

File: R/integration.R (hunk 1 above)
Comment (danielStrobl) [reply R/integration.R:47]: ok, I'll try that. The current setupd doesn't work either way

**Comment by danielStrobl** [comment]
ok, I got it working in some cases now.
For the full_feature cases, I get this error here:
```
Error in nn2(data = c(1.57559389046269, -1.48019171953529, 0.398490401872583,  : 
  Cannot find more nearest neighbours than there are points
Calls: runSeuratRPCA ... RunIntegration -> FindWeights -> NNHelper -> do.call -> nn2
```

**Comment by LuckyMD** [comment]
Looks good now! The above error is something that is Seurat internal. We ran into that a couple of times. It has to do with not finding enough anchors and can apparently happen with certain dataset sizes. That's a run fail that we can't control, but have to accept. HVG will work better anyway, so i think it's fine.

**Review by danielStrobl — APPROVED** [review]

**Pull request merged**